ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -177,15 +177,28 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			rayJobInstance.Status.DashboardURL = clientURL
 		}
 
+		if rayJobInstance.Spec.SubmissionMode == rayv1.NoneMode {
+			logger.Info("SubmissionMode is NoneMode and the RayCluster is created. Transition the status from `Initializing` to `Waiting`.")
+			rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusWaiting
+			break
+		}
+
 		if rayJobInstance.Spec.SubmissionMode == rayv1.K8sJobMode {
 			if err := r.createK8sJobIfNeed(ctx, rayJobInstance, rayClusterInstance); err != nil {
 				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 			}
 		}
 
-		logger.Info("Both RayCluster and the submitter K8s Job are created. Transition the status from `Initializing` to `Running`.",
+		logger.Info("SubmissionMode is K8sJobMode and both RayCluster and the submitter K8s Job are created. Transition the status from `Initializing` to `Running`.",
 			"RayJob", rayJobInstance.Name, "RayCluster", rayJobInstance.Status.RayClusterName)
 		rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusRunning
+	case rayv1.JobDeploymentStatusWaiting:
+		rayJobId, found := rayJobInstance.ObjectMeta.Annotations[utils.RayJobSubmissionIdLabelKey]
```

Comment:
Should this phase also send a request to the Ray head and check if a Ray job with the submission ID also exists? Otherwise we ciykd transition to Running even if the job was never submitted.

Reply:
I guess we technically do this in Running phase already, what's the expected behavior if the job never gets submitted? Would the RayJob just fail at that point?

Reply:
Done in the latest commit. I tried to get RayJob info with the submission id, and transit it to Running only if there is no error. Tests updated accordingly, too.

Reply:
We may not need to check the `JobStatus` in this state. It is already checked in `JobDeploymentStatusRunning`.

Reply:
We don't need to call `GetJobInfo` here because it is `JobDeploymentStatusRunning`'s responsibilities to check the Ray job status.

https://github.com/ray-project/kuberay/blob/22c2b4597fc88b74350ed305133678d1210b7138/ray-operator/controllers/ray/rayjob_controller.go#L222-L242